docs: dedupe subgraph and connector guides (#88)
aragon/connect: docs/advanced/app-subgraphs.md

# App Subgraphs

Making a subgraph for an aragon app is not much different from making any other kind of subgraph. To learn how to do that, check out [TheGraphs's documentation](https://thegraph.com/docs).

> In this subgraph template, we merely hook up a bunch of Aragon related data sources that will detect any installed organization, app, token, etc in Aragon deployments. This is hidden from you, so that writting the subgraph is as similar as possible as writting a normal subgraph.

Reviewer: ```suggestion
In our app subgraphs, we merely hook up a bunch of Aragon related data sources that will detect any installed organization, app, token, etc in Aragon deployments. This is hidden from you, so that writing the subgraph is as similar as possible as writing a normal subgraph.
```

## Step by step guide for using this guide

To create a new subgraph for Aragon, start off by creating a copy of one of our existing connectors, such as [`connect-thegraph-voting`](https://github.com/aragon/connect/tree/master/packages/connect-thegraph-voting). Then follow the steps below for modifying them to fit your needs.

### 1. Modify package.json with your username and subgraph name

> In `package.json`, you'll find a bunch of scripts that reference the user "aragon" and the app name "voting". Change these to your user or Github organization name.

Reviewer: ```suggestion
In `package.json`, you'll find a bunch of scripts that reference the user "aragon" and the app name "voting". Change these to your user or Github organization name, and your application name.
```

The specification for the nomenclature is: `Aragon <AppName> <Network> [Staging]`.

Staging subgraphs are intended for development, and are subgraphs that index very quickly. More info on this below.

### 3. Set up the subgraph manifest files

The way we handle data sources in these subgraphs is a bit sophisticated, so the `subgraph.yaml` file at the root is actually a generated file and should not be edited. This is why you may not find this file at the root of the project until it is generated first.

Instead, edit `subgraph.template.yaml`. You'll notice that there are a bunch of [Mustache](https://mustache.github.io) tags in this file. This is stuff that you shouldn't have to worry about. Initially, the part that matters to you is whatever is not a mustache tag. Here, you can define static data sources in the `dataSources` section, and dynamic datasources in the `templates` section.

> If your writting a subgraph for an Aragon app, its data source specification should go in the latter, since it's instances will be generated dynamically. As an example, this template declares the "Voting" data source template. This data source will be hooked up to any instances of voting apps found in all deployed Aragon organizations.

Reviewer: ```suggestion
If your writing a subgraph for an Aragon app, its data source specification should go in the latter, since it's instances will be generated dynamically. As an example, this template declares the "Voting" data source template. This data source will be hooked up to any instances of voting apps found in all deployed Aragon organizations.
```

### 4. Provide the ABIs defined in your data source

> As with an TheGraph subgraph, ABIs referenced in your manifest should be provided in the `abis` folder.

Reviewer: ```suggestion
As with any TheGraph subgraph, ABIs referenced in your manifest should be provided in the `abis` folder.
```

Of course, this deployment will take much more time to index, potentially days.

## Troubleshooting

### Some data is missing in my subgraph

Even though templates collect data from many sources, there are some cases in which a contract is deployed in a way that isolates it from the general data sources. For example, you may deploy an organization manually, without using the official templates or factories. If this is the case, such organization will not be detected by the bootstrapped data sources. There may also be bugs in how a template scans the data sources. If you find missing information in your subgraph, please contact us to discuss your case to see if there's a bug in the template. If a bug is not the case, you may have to manually insert specific data sources in the subgraph, such as isolated organizations, apps or tokens.

### I'm getting errors about missing ABIs when the subgraph is indexing

> When a reducer is run, it's run in the context of the data source that defined it. For example, hooks are triggered by `src/base/Kernel.ts` when the `NewAppProxy` event is detected in an Organization. You need to include the missing ABI in manifest/templates/contracts/Kernel.template.yaml for it to be available in this reducer. As a rule of thumb, if your data source will be triggered by a base Aragon data source (Organization templates, Organization factories, etc), include your its ABI in the manifest/templates/contracts files.

Reviewer: ```suggestion
When a reducer is run, it's run in the context of the data source that defined it. For example, hooks are triggered by `src/base/Kernel.ts` when the `NewAppProxy` event is detected in an Organization. You need to include the missing ABI in `manifest/templates/contracts/Kernel.template.yaml` for it to be available in this reducer. As a rule of thumb, if your data source will be triggered by a base Aragon data source (Organization templates, Organization factories, etc), include its ABI in the `manifest/templates/contracts` files.
```

## Troubleshooting

> #### Some data is missing in my subgraph

Reviewer 2: Should this (and the following ones) be a `###` heading?

Author: Done in 127939c!
